### PR TITLE
Adding an accounting module

### LINKED ISSUE #676: Add an accounting module
For being able to offer print as a service, we would need an accounting module. Something that stores (PG DB) what template was printed on what application from what referer.

### PR DESCRIPTION
**Pull request by pvalsecc**
Closes #676

### CODE REVIEW

File: core/src/main/java/org/mapfish/print/output/AbstractJasperReportOutputFormat.java [hunk 1]
```diff
@@ -84,8 +86,8 @@ protected abstract void doExport(final OutputStream outputStream, final Print pr
             throws JRException, IOException;
 
     @Override
-    public final void print(final String jobId, final PJsonObject requestData, final Configuration config,
-                            final File configDir, final File taskDirectory, final OutputStream outputStream)
+    public final Processor.ExecutionContext print(final String jobId, final PJsonObject requestData, final Configuration config,
+                                                  final File configDir, final File taskDirectory, final OutputStream outputStream)
```
Comment (sbrunner) [inline core/src/main/java/org/mapfish/print/output/AbstractJasperReportOutputFormat.java]: Too long lines ...

File: core/src/main/java/org/mapfish/print/output/OutputFormat.java [hunk 2]
```diff
@@ -26,15 +27,14 @@
 
     /**
      * Performs the print and writes to the report in the correct format to the outputStream.
-     *
-     * @param jobId the job ID
+     *  @param jobId the job ID
```
Comment (sbrunner) [inline core/src/main/java/org/mapfish/print/output/OutputFormat.java]: extra space

File: core/src/main/java/org/mapfish/print/output/OutputFormat.java [hunk 3]
```diff
@@ -26,15 +27,14 @@
...
      * @param spec the data from the client, required for writing.
      * @param config the configuration object representing the server side configuration.
      * @param configDir the directory that contains the configuration, used for resolving resources like images etc...
      * @param taskDirectory the temporary directory for this printing task.
      * @param outputStream the stream to write the result to
      */
-    void print(String jobId, PJsonObject spec, Configuration config, File configDir, File taskDirectory,
-               OutputStream outputStream) throws Exception;
+    Processor.ExecutionContext print(String jobId, PJsonObject spec, Configuration config, File configDir, File taskDirectory,
```
Comment (sbrunner) [inline core/src/main/java/org/mapfish/print/output/OutputFormat.java]: too long line

File: core/src/main/java/org/mapfish/print/processor/ExecutionStats.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+public class ExecutionStats {
+    private List<MapStats> mapStats = new ArrayList<>();
+
+    /**
+     * Add statistics about a created map.
+     *
+     * @param mapContext the
+     * @param mapValues the
+     */
+    public synchronized void addMapStats(final MapfishMapContext mapContext, final MapAttribute.MapAttributeValues mapValues) {
```
Comment (sbrunner) [inline core/src/main/java/org/mapfish/print/processor/ExecutionStats.java]: same

File: core/src/main/java/org/mapfish/print/output/AbstractJasperReportOutputFormat.java (hunk 1 above)
Comment (pvalsecc) [reply core/src/main/java/org/mapfish/print/output/AbstractJasperReportOutputFormat.java]: Will fix the long line in another PR. The plan is to have IDEA auto-format all the files once for good.

File: core/src/main/java/org/mapfish/print/output/OutputFormat.java (hunk 2 above)
Comment (pvalsecc) [reply core/src/main/java/org/mapfish/print/output/OutputFormat.java]: Fixed

**Pull request merged**